symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/installAddonAPIAction.class.php

```diff
@@ -57,28 +58,60 @@ public function execute($request)
     /**
      * @param $addonURL
      * @param $addonDetail
+     * @return int
      * @throws CoreServiceException
      * @throws DaoException
      */
     private function getAddonFile($addonURL, $addonDetail)
     {
-        $addonfile = $this->getApiManagerService()->getAddonFile($addonURL);
-        return $this->installAddon($addonfile, $addonDetail);
+        $addonFilePath = $this->getApiManagerService()->getAddonFile($addonURL, $addonDetail);
+        return $addonFilePath;
     }
 
     /**
-     * @param $addon
+     * @param $addonFilePath
      * @param $addonDetail
+     * @return bool
      * @throws DaoException
+     * @throws Doctrine_Connection_Exception
+     * @throws Doctrine_Transaction_Exception
      */
-    protected function installAddon($addon, $addonDetail)
+    protected function installAddon($addonFilePath, $addonDetail)
     {
+        $pluginname = $this->getMarcketplaceService()->extractAddonFile($addonFilePath);
+        $currentDirectory = dirname(__FILE__);
+        $symfonyPath = $currentDirectory . "/../../../../../";
```

Comment:
Use `sfConfig::get('sf_root_dir')` instead of `$currentDirectory . "/../../../../../"`.